draft-ietf-ace-edhoc-oscore-profile.md

```diff
@@ -515,23 +508,17 @@ That is, C sends a POST request to the /authz-info endpoint at RS, with the requ
 
 The communication with the /authz-info endpoint does not have to be protected, except for the update of access rights case described below.
 
-In case of initial access token provisioning to RS for this Client, or in other cases without a valid EDHOC session, the uploading of the access token is followed by the execution of the EDHOC protocol (or combined using EAD as described in {{edhoc-exec}}) and by the derivation of an OSCORE Security Context, as detailed later in this section.
-
-In the following, we outline some alternative processing, which occur when the provisioned access token or the established OSCORE Security Context for various reasons is no longer available or sufficient. In the cases below, it is assumed that the EDHOC session is still valid, otherwise the processing essentially falls back to the case discussed above.
-
-1. C may be required to re-POST the access token, since RS may have deleted a stored access token (and associated OSCORE Security Context) at any time, for example due to all storage space being consumed. This situation can be detected by C when it receives a 4.01 (Unauthorized) response from RS, especially as an "AS Request Creation Hints" message (see {{Section 5.3 of RFC9200}}.
+The Client provisioning an initial access token to the RS is followed by the execution of the EDHOC protocol (or combined using EAD as described in {{edhoc-exec}}) and by the derivation of an OSCORE Security Context, as detailed later in this section.
 
-2. C may be posting the first access token in a new series, e.g., because the old access token expired and thus its series terminated.
+The same procedure applies in other cases without a valid OSCORE Security Context shared between the C and RS, for example:
 
-3. C may need to update the OSCORE Security Context, e.g., due to a policy limiting its use in terms of time or amount of processed data, or to the imminent exhaustion of the OSCORE Sender Sequence Number space. The OSCORE Security Context can be updated by:
+* The old access token has expired (and thus its series terminated and associated OSCORE Security Context deleted) so C provisions a new access token to RS.
 
-    a. using the KUDOS key update protocol specified in {{I-D.ietf-core-oscore-key-update}}, if supported by both C and RS; or
+* The Client re-POSTs an access token which was deleted by RS (and thus its associated OSCORE Security Context) for example due to lack of storage. This situation can be detected by C when it receives a 4.01 (Unauthorized) response from RS, e.g., as an "AS Request Creation Hints" message, see {{Section 5.3 of RFC9200}}.
 
```

Comment:
Another reason for C or RS to delete the OSCORE Security Context is the EDHOC session having become invalid (e.g., due to the expiration of an authentication credential), with consequent deletion of the application keying material derived from it.

Reply:
I will add EDHOC session becoming invalid to the set of reasons.

Reply:
The first list above includes reasons to invalidate a Security Context, which is thus supposed to be also deleted. The second list gives one further reason to delete the Security Context (lack of storage). Can the two lists be merged into a single, main one?

The focus can be on reasons to delete a Security Context. These are: i) the Security Context has become invalid, which in turn happens if, e.g., the token has expired, the EDHOC session has become invalid, other policies have determined so; ii) lack of storage at the RS and consequent deletion at the RS.